ulid.go

```diff
@@ -454,6 +454,18 @@ func (id ULID) Compare(other ULID) int {
 	return bytes.Compare(id[:], other[:])
 }
 
+// Copy sets id equal to the value of other.
+func (id *ULID) Copy(other ULID) {
```

Comment:
It's confusing that a method named Copy would mutate the receiver — Copy should return a ULID value.

Reply:
I'm implementing a third-party unmarshalling interface. Returning a value doesn't work, mutating an existing struct field is basically a requirement. However, as I was working on a simplified example I realized how to make my implementation sufficiently readable as is.

You have a good point about mutating the caller being surprising. I could either remove the method, or replace it with a function: `Copy(dst &ULID, src ULID)`, if you think it could be helpful to future users.

I assume `IsZero()` is acceptable?

While neither change is strictly necessary, I do feel they improve readability and reduce coupling to the underlying byte array implementation.

Reply:
Actually, on reflection, I don't think IsZero is a good idea, because a zero value ULID is still a valid ULID.

Reply:
That's an odd opinion given IsZero isn't about invalid / valid but instead default / non-default, which users can choose to interpret as uninitialized / initialized.

As the system implementer, isn't it my prerogative to decide if the zero value is valid or not in my system?